### PR TITLE
Rewrite NettyResponseFuture using CompletableFuture

### PR DESCRIPTION
**Pull request by stepancheg**
`CompletableFuture` seems to be as expensive as `CountDownLatch`, however, when user asks for `toCompletableFuture`, `NettyResponseFuture` now returns already allocated `CompletableFuture` object without `Runnable` registration in `AbstractListenableFuture`.

Also, I believe that `CompletableFuture` is well optimized.

Tests pass, although it is hard to tell if anything is broken, because `NettyResponseFuture` implementation is hard.

### CODE REVIEW

**Comment by slandelle** [comment]
Looking good! Please let me review carefully later today.

File: client/src/main/java/org/asynchttpclient/netty/NettyResponseFuture.java [hunk 1]
```diff
@@ -47,24 +45,18 @@
  * 
  * @param <V> the result type
  */
-public final class NettyResponseFuture<V> extends AbstractListenableFuture<V> {
+public final class NettyResponseFuture<V> implements ListenableFuture<V> {
```
Comment (slandelle) [inline client/src/main/java/org/asynchttpclient/netty/NettyResponseFuture.java]: So `AbstractListenableFuture` and `RunnableExecutorPair` can be removed, as they are replaced with `CompletableFuture#whenCompleteAsync`, right?

Comment (stepancheg) [inline client/src/main/java/org/asynchttpclient/netty/NettyResponseFuture.java]: Yes. Should I remove it?

Comment (slandelle) [inline client/src/main/java/org/asynchttpclient/netty/NettyResponseFuture.java]: Yes, please

**Comment by stepancheg** [comment]
Update PR with removed `AbstractListenableFuture` and `RunnableExecutorPair`.

**Comment by slandelle** [comment]
Thanks a lot!

**Pull request merged**